src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -5149,13 +5150,26 @@ protected sealed override ImmutableArray<ISymbol> GetDeclaredSymbolsCore(SyntaxN
                 return this.GetDeclaredSymbols(field, cancellationToken);
             }
 
-            var symbol = GetDeclaredSymbolCore(declaration, cancellationToken);
-            if (symbol != null)
```

Comment:
the caller of this method is at: https://github.com/dotnet/roslyn/blob/50aa37f11d39a55f70254eca281749dc40cb52ee/src/Compilers/Core/Portable/DiagnosticAnalyzer/SuppressMessageAttributeState.cs#L166

This code walks upwards, looking at the nodes it hits and seeing what attributes those nodes have.  HOwever, this goes awry when hitting a named type *node* as the node maps to a named-type *symbol* which does not have the attributes anymore.  By updating this code to return both the named-type and PC we can see the entire set of attributes (similar to how the existing code can handle field-decls vs field-symbols).